add: Capitalism

Pelmenik123 opened the pull request:
## Описание

Добавляет зарплаты, изменяет оплату карго и вобще делает все во класс

## Ссылка на предложение/Причина создания ПР

https://discord.com/channels/617003227182792704/755125334097133628/1281882574330789959

## Демонстрация изменений

Добавляет новую подсистему капитализма, добавляет доп проки для убоства, все подробно в комментариях

## Тесты

Ну... побегал потыкал заказы выполнял.

Pelmenik123 committed:
Капиталистические свиньи в восторге

Pelmenik123 committed:
Update capitalism.dm

забыл изменить это перед созданием пулла

Den109G commented on code/controllers/subsystem/capitalism.dm:
```diff
@@ -0,0 +1,165 @@
+
+
+	//ВНИМАНИЕ. Статистика для гринтекста
+	//А зачем я табы сделал....
```
Тот же вопрос. Почему не пробелы?

Pelmenik123 replied:
поправлю

Den109G replied:
Не поправил.

Pelmenik123 replied:
врешь не уйдешь, правил это

Pelmenik123 committed:
Update Accounts.dm

Забыл отступ сделать

Den109G commented on code/controllers/subsystem/capitalism.dm:
```diff
@@ -0,0 +1,165 @@
+
+	if(default_counter > 300)
+		GLOB.priority_announcement.Announce("Станция признана убыточным объектом. Хорошего дня.", "Расторжение контрактов.", 'sound/AI/commandreport.ogg')
+		set_security_level(SEC_LEVEL_EPSILON)
```
Простите, а где в ТЗ код Эпсилон хотя бы упомянут?

Pelmenik123 replied:
Нигде, выпадение невозможно

NightDawnFox replied:
технически эта часть кода вообще работать не будет из-за того, что по измененному ТЗ деньги берутся из воздуха и дефолта вообще не будет

Pelmenik123 replied:
Берутся из воздуха на время тестов

Den109G replied:
> Нигде, выпадение невозможно

~~А тогда зачем...~~
Закомменть весь связанный код, если он не должен работать.

Pelmenik123 replied:
ну блина(((((

Den109G commented on code/controllers/subsystem/capitalism.dm:
```diff
@@ -0,0 +1,165 @@
+#define FREQUENCY_SALARY 5 MINUTES
+#define TEN_THOUSAND_CREDITS 10000 //Хз честно как иначе назвать roflcat
```
Может что-то "`CREDITS_TO_STOP_DEFAUL`"?

Den109G replied:
А и да. `#undef` если нигде не используется после.

Pelmenik123 replied:
Не совсем так... 10000 это просто рандомное число с потолка. Заменю на Extra_money

Den109G commented on code/controllers/subsystem/capitalism.dm:
```diff
@@ -0,0 +1,165 @@
+		base_account.credit(total_station_goal_bounty, "Начисление награды за выполнение цели.", "Отдел развития Нанотрейзен", base_account.owner_name)
+
+//status - TRUE/FALSE
+/datum/controller/subsystem/capitalism/proc/default_annonce(var/status)
```
Зачем мы вообще передаём переменную? Разве нельзя это сделать на `default_status`?
```dm
if(default_status)
   to_chat(world, "Денег малА")
else
   GLOB.priority_announcement.Announce("Всё ок")
```

Pelmenik123 replied:
Ладно это было решение шизы

Den109G commented on code/controllers/subsystem/non-firing/cargo_quests.dm:
```diff
@@ -2,6 +2,13 @@
+
+//Все больше вчитываясь в строки я понимаю, ох зря я сюда полез.
+
+//ДАЛЬШЕ БОГА НЕТ
```
Комменты прошу на английском.

Pelmenik123 replied:
Рофло коменты оставлю на русском

NightDawnFox replied:
Нет, не оставишь. Рофлокомменты либо на английский, либо удаляем. За приколы ломаем колени.

Pelmenik123 commented:
Никаких конфликтов, каеф

Den109G commented on code/controllers/subsystem/non-firing/cargo_quests.dm:
```diff
@@ -195,25 +214,33 @@ SUBSYSTEM_DEF(cargo_quests)
 	min_quest_time = 20
 	max_quest_time = 30
 	for_easy_mode = TRUE
 
```
Зачем нам пробелы между переменными одного квеста? Ко всем изменениям относится.
```suggestion
```

Pelmenik123 replied:
хотел выделить типо какое важное значение БАУНТИ

Den109G commented on code/game/jobs/job/job.dm:
```diff
@@ -65,9 +65,14 @@
 	var/spawn_ert = 0
 	var/syndicate_command = 0
 
+	//legacy_system 
```
If we don't use it - why leave it? Ну и из-за коммента кажется что всё до `//////////` - это остатки прошлого.

Pelmenik123 replied:
Ладно заменю на start_legact end_legacy

Den109G replied:
Зачем вообще оставлять, если не используется? Если используется - это не legacy.

Pelmenik123 replied:
Я не хотел искать используется это где то или нет
Поэтому оставил чтобы не было ошибок

Den109G commented on code/modules/economy/Accounts.dm:
```diff
@@ -3,15 +3,31 @@
+		GLOB.next_account_number = rand(111111, 999999)
+
+		GLOB.trull_account = new()
+		GLOB.trull_account.owner_name = "Account of the personnel department of the Central Command"
```
Если это ЦК - так и надо писать. Не на всех картах кораблик является ЦК.
Убери `trull_account` тролля и поставь `CC_account`

Pelmenik123 replied:
Это вобще никто не увидит

Den109G replied:
Я увидел. Чини.

NightDawnFox commented on code/controllers/subsystem/capitalism.dm:
```diff
@@ -0,0 +1,165 @@
+#define FREQUENCY_SALARY 5 MINUTES
+#define TEN_THOUSAND_CREDITS 10000 //Хз честно как иначе назвать roflcat
```
меняй все комментарии на английский. Русские комментарии у нас неприемлемы и являются остатком двух-трехлетней давности. Или ошибками

Pelmenik123 replied:
((((((

NightDawnFox commented on code/controllers/subsystem/capitalism.dm:
```diff
@@ -0,0 +1,165 @@
+	name = "Capitalism"
+	ss_id = "capitalism_subsystem"
+	init_order =  INIT_ORDER_CAPITALISM
+	offline_implications = "Выплаты зарплат приостановлены, по идеи выплаты за задания карго не сломаются. Награда за цель не выплачивается. Немедленных действий не требуется."
```
```suggestion
	offline_implications = "Выплаты зарплат приостановлены, по идее выплаты за задания карго не сломаются. Награда за цель не выплачивается. Немедленных действий не требуется."
```

Den109G commented on code/modules/economy/Accounts_DB.dm:
```diff
@@ -25,16 +25,8 @@ GLOBAL_VAR(current_date_string)
-
+	// TODO выполнено, SScapitalism
+	
 	if(!GLOB.current_date_string)
```
По хорошему, и это перенести в сабсистему, раз говоришь что выполнил просьбу АА. `date system breaks` если нет этой консоли.

Pelmenik123 replied:
Продублирую и в саб систему

Den109G commented on code/modules/economy/quests/thing_quests.dm:
```diff
@@ -1,3 +1,4 @@
+//Оставь надежду надежду всяк сюда входящий
```
Этот комментарий повторяется десять раз. Код выглядит углажено, что не так?

Pelmenik123 replied:
Я сильно запутался когда хотел полностью поменять выплаты, не осилил, умер

Den109G replied:
Лучше убрать, раз всё работает как нужно и без костылей.

NightDawnFox commented on code/game/gamemodes/scoreboard.dm:
```diff
@@ -353,7 +353,14 @@ GLOBAL_VAR(scoreboard) // Variable to save the scoreboard string once it's been
-	<b>Клоуна избили:</b> [score_clown_abuse] [declension_ru(score_clown_abuse, "раз", "раза", "раз")]<br><br>"}
+	<b>Клоуна избили:</b> [score_clown_abuse] [declension_ru(score_clown_abuse, "раз", "раза", "раз")]<br><br>
+	
+	<U>Капитализм статистика</U><br>
```
название не очень

Pelmenik123 replied:
Аминь, да будет Статистика финансов

NightDawnFox replied:
Может Финансовая статистка?

Pelmenik123 replied:
Ладно

NightDawnFox commented on code/game/jobs/job/engineering.dm:
```diff
@@ -21,9 +21,12 @@
-	money_factor = 7
 	outfit = /datum/outfit/job/chief_engineer
 
+	salary = 300
```
Почему ты ввел дефайн для зарплаты ассистентам, но не сделал то же самое для остальных профессий?

Pelmenik123 replied:
Просьба от https://discord.com/channels/617003227182792704/734823601110515882/1289658495892717588

Pelmenik123 committed:
Update Accounts.dm

Меняю раз

Pelmenik123 committed:
Update capitalism.dm

распили меня болгаркой

Pelmenik123 committed:
Update capitalism.dm

пупупу

Pelmenik123 committed:
Update scoreboard.dm

Финансовая статистка

Pelmenik123 committed:
Update capitalism.dm

Бобер курва

Pelmenik123 committed:
Update cargo_quests.dm

пендофицикация

Pelmenik123 committed:
Update cargo_quests.dm

sdsds

Pelmenik123 committed:
Update thing_quests.dm

ыфвфвы

Pelmenik123 committed:
Update Accounts_DB.dm

// TODO done, SScapitalism

Pelmenik123 committed:
Update capitalism.dm

// TODO done, SScapitalism

Pelmenik123 committed:
Update card.dm

пробелы апдейт

Pelmenik123 committed:
Update security.dm

в

Pelmenik123 committed:
Update support.dm

//QM IS NOT THE HEAD!!

Pelmenik123 committed:
Update support_chaplain.dm

//God will not give you a salary roflcat

Pelmenik123 committed:
Update centcomm_departaments.dm

15 percent on the account of the beggars

Pelmenik123 committed:
Update Accounts.dm

пробел ликвидирован

Pelmenik123 committed:
Update reagents_quests.dm

ыввы

Pelmenik123 commented:
Это немного, но это честная работа

Pelmenik123 committed:
Update job.dm

наследие

Pelmenik123 committed:
Update capitalism.dm

убирал to_chat

Pelmenik123 committed:
Update utils.dm

For revenue statistics from the vending machine

DefaultMihail commented:
QA приключение на 5 минут:
1. При зачислении ЗП она не снимается со счёта станции, она идёт с "Account of the personnel department of the Central Command" а не с "NSS Kerberos Station Account"
2. При спауне нет кошелька с наличкой
3. Дефолт не обнаружен, снял все деньги со счёта станции и дождался пополнения ЗП
4. Построил днк хранилище и получил на инженерах за постройку целое ничего

Pelmenik123 commented:
с днк действительно баг, остальное фича

Den109G commented on code/game/jobs/job/job.dm:
```diff
@@ -65,9 +65,14 @@
 	var/syndicate_command = 0
 
+	//legacy_system_start
 	var/money_factor = 1 // multiplier of starting funds
 	var/random_money_factor = FALSE // is miltiplier randomized (from 4x to 0.25x for now)
-
+	//legacy_system_end
```
Я по коду вижу, что эти переменные не используются. Удали.

Pelmenik123 committed:
Update capitalism.dm

бобр курва

Pelmenik123 committed:
Update station_goal.dm

пупупу

Pelmenik123 committed:
Update capitalism.dm

job_titles[account.linked_job.title]

Pelmenik123 committed:
Update capitalism.dm

jobs_payment

Pelmenik123 committed:
Update capitalism.dm

jobs_payment[account.linked_job.title]

Pelmenik123 reviewed (commented):
Сделал что ден говорил

FanTik-Ratge commented:
Срать вечно 🐀🤙

Den109G commented on code/modules/economy/quests/_base_quests.dm:
```diff
@@ -152,7 +152,11 @@
+	
+	var/cargo_quest_reward = 0 			//Награда за квест, считайте дебаг залупа roflcat
+	var/list/bounty_jobs = list() 		//Должности которым будет идти зарплата. (Неееет я не буду делать часть этого в new)
+	var/linked_deportament = "Cargo" 	//Указывается ключ депортамента чтобы брать его из глобального списка, нет я не сделаю загрузку в new, боюсь чет сломать
```
Полиция словаря. Нет депорта у нас.
```suggestion
	var/linked_departament = "Cargo" 	//Указывается ключ депортамента чтобы брать его из глобального списка, нет я не сделаю загрузку в new, боюсь чет сломать
```

Den109G commented on code/controllers/subsystem/capitalism.dm:
```diff
@@ -0,0 +1,175 @@
+
+		if(account.salary_payment_active && account.linked_job.salary && !account.suspended)
+
+			if(payment_account.charge(account.linked_job.salary, account, "Выплата зарплаты персоналу.", "Nanotrasen personal deportament" , payment_account.owner_name, payment_account.owner_name , payment_account.owner_name))
```
Deportment - личность. Department - отдел.
```suggestion
			if(payment_account.charge(account.linked_job.salary, account, "Выплата зарплаты персоналу.", "Nanotrasen personal departament" , payment_account.owner_name, payment_account.owner_name , payment_account.owner_name))
```

Den109G commented on code/controllers/subsystem/capitalism.dm:
```diff
@@ -0,0 +1,175 @@
+	return
+
+// In short, as for beggars, but for departments
+/datum/controller/subsystem/capitalism/proc/smart_deportament_payment(var/list/keys_deportament, var/money)
```
```suggestion
/datum/controller/subsystem/capitalism/proc/smart_departament_payment(var/list/keys_departament, var/money)
```

Den109G commented on code/modules/economy/quests/centcomm_departaments.dm:
```diff
@@ -164,14 +171,31 @@
+
+/datum/quest_customer/corp/send_reward(reward, var/list/copmpleted_quests = list())
+	var/list/nishebrod_jobs = list()
+	var/list/linked_deportaments = list() //HEHE HI HA
```
```suggestion
	var/list/linked_departaments = list() //HEHE HI HA
```

DefaultMihail commented on code/controllers/subsystem/capitalism.dm:
```diff
@@ -0,0 +1,175 @@
+	for(var/datum/station_goal/goal in SSticker.mode.station_goals)
+		if(!goal)
+			continue
+		if(goal.check_completion() && !(goal in complited_goals))
```
Ты проверяшь тут ВЫПОЛНЕНИЕ цели, а не её постройку. На примере того же ДНК, он ставит check_completion, только когда днк заполнено.

Pelmenik123 replied:
пупупу, тогда прийдется отдельный прок писать, пусть будет фича

Pelmenik123 replied:
фича

Pelmenik123 commented:
Жду апрув день 1

Pelmenik123 committed:
Update job.dm

ыввывы

Pelmenik123 committed:
Update security.dm

money_factor

Pelmenik123 committed:
Update science.dm

money_factor

Pelmenik123 committed:
Update code/modules/economy/quests/_base_quests.dm

да

Co-authored-by: Den109G <87372121+Den109G@users.noreply.github.com>

Pelmenik123 committed:
Update code/controllers/subsystem/capitalism.dm

лд

Co-authored-by: Den109G <87372121+Den109G@users.noreply.github.com>

Pelmenik123 committed:
Update code/controllers/subsystem/capitalism.dm

yes

Co-authored-by: Den109G <87372121+Den109G@users.noreply.github.com>

Pelmenik123 committed:
Update centcomm_departaments.dm

linked_deportament

Pelmenik123 committed:
Update reagents_quests.dm

linked_deportament

Pelmenik123 committed:
Update thing_quests.dm

linked_deportament

Pelmenik123 committed:
Update centcomm_departaments.dm

smart_deportament_payment

Pelmenik123 committed:
Update capitalism.dm

 keys_deportament

Pelmenik123 committed:
Update capitalism.dm

 keys_deportament

Pelmenik123 committed:
Update code/controllers/subsystem/capitalism.dm

да

Co-authored-by: NightDawnFox <116907157+NightDawnFox@users.noreply.github.com>

Vladisvell commented on code/controllers/subsystem/capitalism.dm:
```diff
@@ -0,0 +1,175 @@
+/datum/controller/subsystem/capitalism/proc/payment_process()
+	. = TRUE
+
+	for(var/datum/money_account/account in GLOB.all_money_accounts)
+
+		if(account.salary_payment_active && account.linked_job.salary && !account.suspended)
+
+			if(payment_account.charge(account.linked_job.salary, account, "Выплата зарплаты персоналу.", "Nanotrasen personal departament" , payment_account.owner_name, payment_account.owner_name , payment_account.owner_name))
+
+				account.notify_pda_owner("<b>Поступление зарплаты </b>\"На ваш привязанный аккаунт поступило [account.linked_job.salary] кредитов\" (Невозможно Ответить)", FALSE)
+				total_salary_payment += account.linked_job.salary
+
+			else
+				return FALSE
```
зачем

так

много

пустых строк

Pelmenik123 replied:
я художник я так вижу

Pelmenik123 replied:
привычка разбивать так на "блоки кода"

Pelmenik123 replied:
+ так легче находить и каши не будет

Pelmenik123 committed:
Update capitalism.dm

пендосия апдейт

Pelmenik123 committed:
Update cash.dm

A crutch for the vending machine, well, or an ingenious roflcat solution

Pelmenik123 committed:
Update _base_quests.dm

The department key is specified to take it from the global list, no, I will not upload to new, I'm afraid to break even

Pelmenik123 committed:
Update quest_console.dm

Abandon hope, everyone who enters here

Pelmenik123 committed:
Update support.dm

Abandon hope, everyone who enters here

Pelmenik123 committed:
Update capitalism.dm

What kind of money for each department

Pelmenik123 committed:
Update capitalism.dm

The counter for the number of defaults, I definitely won't make a joke

Pelmenik123 committed:
Update security.dm

Pelmenik123 committed:
Update silicon.dm

Pelmenik123 committed:
Update centcomm_departaments.dm

If not, it pays to the account of the department

Pelmenik123 committed:
Update support.dm

Pelmenik123 committed:
Update supervisor.dm

Pelmenik123 committed:
Update support.dm

все апруверы выписаны из русских

FanTik-Ratge commented:
Спрайты не в стиле парадиза 🐀👑

Pelmenik123 commented:
фантик это френдли фаер, ты себе в колено выстрелил

Pelmenik123 commented:
Ден умоляю, дай мне этот апрув. Мержани его на мейны сразу

Pelmenik123 committed:
Update capitalism.dm

отступы апдейт

Pelmenik123 committed:
Update capitalism.dm

фикс на упреждение

Pelmenik123 commented:
гооол

Pelmenik123 committed:
Update capitalism.dm

во

Pelmenik123 committed:
Update Accounts_DB.dm

Охота туду

Pelmenik123 committed:
Update capitalism.dm

Den109G commented on code/modules/economy/quests/quest_console.dm:
```diff
@@ -1,5 +1,6 @@
 
 #define PRINT_COOLDOWN 10 SECONDS
+//Abandon hope, everyone who enters here
```
Не вижу смысла в комментарии. Особенно раз ничего в нём не менял.
```suggestion
```

Den109G commented on code/modules/shuttle/supply.dm:
```diff
@@ -207,13 +207,15 @@
 
 					var/cost = tech.getCost(SSshuttle.techLevels[tech.id])
+					if(tech.level >= 7)
+						SScapitalism.base_account.credit(7000, "Благодарность за вклад в науку.", "Nanotreisen Institute terminal#[rand(111,333)]", "Nanotreisen Institute")
```
```suggestion
						SScapitalism.base_account.credit(7000, "Благодарность за вклад в науку.", "Nanotrasen Institute terminal#[rand(111,333)]", "Nanotrasen Institute")
```

Pelmenik123 committed:
Update capitalism.dm

Не используется

Pelmenik123 committed:
Update code/modules/shuttle/supply.dm

Распили меня болгаркой

Co-authored-by: Den109G <87372121+Den109G@users.noreply.github.com>

Pelmenik123 committed:
Update code/modules/economy/quests/quest_console.dm

Ладно

Co-authored-by: Den109G <87372121+Den109G@users.noreply.github.com>

Pelmenik123 committed:
Update capitalism.dm

Забыл

Pelmenik123 committed:
Update capitalism.dm

пробелы апдейт

Pelmenik123 committed:
Update capitalism.dm

ывывы

Pelmenik123 committed:
Update capitalism.dm

пробелы апдейт

Pelmenik123 committed:
Update capitalism.dm

пойду выпью яду

Den109G reviewed (dismissed):
Всё по ТЗ. Вопросы решены. Автор ПРа, перестань писать каждый час об этом.

Pelmenik123 commented:
:catrose:

Pelmenik123 committed:
Update subsystems.dm

Авось поможет

Pelmenik123 committed:
Update cash.dm

еыыв

Pelmenik123 committed:
Update cash.dm

ыввы

Pelmenik123 committed:
Update cash.dm

ыввы

Den109G commented on code/game/objects/items/weapons/cash.dm:
```diff
@@ -88,6 +88,13 @@
 	amount = 1000000
 
+//A crutch for the vending machine, well, or an ingenious roflcat solution
+/obj/item/stack/spacecash/magic_linked/New(loc, amt = 150)
```
Нет причин это в `/New` делать.
```suggestion
/obj/item/stack/spacecash/magic_linked/Initialize(mapload, new_amount, merge = TRUE)
```

Pelmenik123 replied:
я копирывал это с USSP кредит

Pelmenik123 committed:
Update code/game/objects/items/weapons/cash.dm

да

Co-authored-by: Den109G <87372121+Den109G@users.noreply.github.com>

Pelmenik123 committed:
Update cash.dm

ывывыв

Pelmenik123 committed:
Update cash.dm

ыввывывы

Pelmenik123 commented:
Готово

Pelmenik123 commented:
Протестил все работает

Pelmenik123 committed:
Update security.dm

фикс

Pelmenik123 committed:
Update science.dm

фикс

Pelmenik123 committed:
Update capitalism.dm

сука

ROdenFL commented on code/controllers/subsystem/capitalism.dm:
```diff
@@ -0,0 +1,160 @@
+	name = "Capitalism"
+	ss_id = "capitalism_subsystem"
+	init_order =  INIT_ORDER_CAPITALISM
+	offline_implications = "Выплаты зарплат приостановлены, по идее выплаты за задания карго не сломаются. Награда за цель не выплачивается. Немедленных действий не требуется."
```
Сидят мужик и читают, что по идее не сломается 😊

Pelmenik123 replied:
ну если сабсистему не удалит то не сломается

Pelmenik123 committed:
Update subsystems.dm

сука влад

Pelmenik123 committed:
Update scoreboard.dm

статистик апдейт
